lib/ftp_session.dart

```diff
@@ -77,10 +110,10 @@ class FtpSession {
     int port = dataListener!.port;
     int p1 = port >> 8;
     int p2 = port & 0xFF;
-    String address = controlSocket.address.address.replaceAll('.', ',');
-
+    // String address = controlSocket.address.address.replaceAll('.', ',');
     // I'm not sure what happened here, the client shows nothing if I comment out this line.
-    await Future.delayed(const Duration(microseconds: 100));
+    // await Future.delayed(const Duration(microseconds: 0));
+    var address = (await _getIpAddress()).replaceAll('.', ',');
```

Comment:
what is the difference that the _getIpAddress make? since `String address = controlSocket.address.address.replaceAll('.', ',');` is alot cleaner.

Reply:
I not sure, I test 3:
```
1. String address = controlSocket.address.address.replaceAll('.', ',');

2. await Future.delayed(const Duration(microseconds: 100));
   String address = controlSocket.remoteAddress.address.replaceAll('.', ',');

3. await Future.delayed(const Duration(microseconds: 100));
   var address = (await _getIpAddress()).replaceAll('.', ',');
```
2 and 3 seem to respond with the same data.: `227 Entering Passive Mode (192,168,127,100,223,209)`, but linux show `Passive mode address mismatch.`

linux terminal context:
```
# _getIpAddress
ftp> ls
227 Entering Passive Mode (192,168,127,100,225,105)
150 Opening data connection
-rw-r--r-- 1 ftp ftp 8196 Jun 02 04:52 .DS_Store
drwxr-xr-x 1 ftp ftp 96 Jun 02 00:44 sefes12
drwxr-xr-x 1 ftp ftp 96 Jun 02 01:14 -a
-rw-r--r-- 1 ftp ftp 12410884 Jan 12 00:45 whisper-release.zip
-rw-r--r-- 1 ftp ftp 7787296 Jan 12 00:50 whisper-release.tar.xz
drwxr-xr-x 1 ftp ftp 96 Jun 02 00:44 未命名文件夹
-rw-r--r-- 1 ftp ftp 25182269 Jan 12 00:33 whisper-release.apk
drwxr-xr-x 1 ftp ftp 96 Jan 12 00:44 whisper.app
226 Transfer complete

# address or remoteAddress
ftp> ls
227 Entering Passive Mode (192,168,127,107,225,138)
Passive mode address mismatch.
ftp> ls
227 Entering Passive Mode (0,0,0,0,226,168)
Passive mode address mismatch.
ftp>
```

Reply:
Since you are testing using ftp command line , I will add the test cases to GitHub ci, 

The reason I didn't add them is that store file doesn't seem to work using the command line ftp on macos, it's just hangs 

If you tried it let me know if I implemented it wrongly on the test case